### ISSUE TITLE
Build failed when using sandboxlib 'chroot' backend

### ISSUE THREAD

**Issue by ssssam** [devcurmudgeon/ybd#84]
```
2015-06-16 16:03:45 [stage2-make] Creating cache artifact
2015-06-16 16:03:48 [stage2-make] Now cached as stage2-make.465d2cf446fc94e418d2aa7a1fa81eefeefba3b4901f053b27e3ec7f0f0df3bf
2015-06-16 16:03:49 [stage2-make] Cleaned up /src/tmp/tmpMl8mo1
2015-06-16 16:03:49 [stage2-make] Elapsed time 00:01:07
2015-06-16 16:03:49 [fhs-dirs] Start build
2015-06-16 16:03:49 [fhs-dirs] Git checkout baserock:baserock/fhs-dirs in /src/tmp/tmp8Rj2P_/fhs-dirs.build
2015-06-16 16:03:50 [fhs-dirs] Upstream version 1218cbd3 (No tag found)
2015-06-16 16:03:50 [fhs-dirs] Logging build commands to /src/cache/ybd-artifacts/fhs-dirs.36a458d5e8ac5aeb6d852142cd40c42a125aa338b2d85f9df5a01bd9c56b1b4d.build-log
2015-06-16 16:03:50 [fhs-dirs] Running install-commands
2015-06-16 16:03:50 [fhs-dirs] Running command:
sh ./create-fhs-dirs "$DESTDIR"
WARNING: Unable to set filesystem_writable_paths=['fhs-dirs.build', 'fhs-dirs.inst', '/dev', '/proc', '/tmp'] in a chroot sandbox, falling back to filesystem_writable_paths=all
2015-06-16 16:03:50 [fhs-dirs] Elapsed time 04:49:32
2015-06-16 16:03:50 [build-essential] Elapsed time 04:49:32
2015-06-16 16:03:50 [strata-core-wrt.morph] Elapsed time 04:49:32
2015-06-16 16:03:50 [firewall3] Elapsed time 04:49:32
2015-06-16 16:03:50 [cpe-wrt] Elapsed time 04:49:32
2015-06-16 16:03:50 [minimal-system-mips64b-openwrt] Elapsed time 04:49:32
2015-06-16 16:03:50 [TOTAL] Elapsed time 04:50:33
2015-06-16 16:03:50 [systems/minimal-system-mips64b-openwrt.morph] Finished
Traceback (most recent call last):
  File "/src/ybd/ybd.py", line 59, in <module>
    with app.timer('CACHE-KEYS', 'Calculating'):
  File "/src/ybd/assembly.py", line 126, in assemble
    assemble(defs, subcomponent)
  File "/src/ybd/assembly.py", line 126, in assemble
    assemble(defs, subcomponent)
  File "/src/ybd/assembly.py", line 118, in assemble
    assemble(defs, dependency)
  File "/src/ybd/assembly.py", line 118, in assemble
    assemble(defs, dependency)
  File "/src/ybd/assembly.py", line 126, in assemble
    assemble(defs, subcomponent)
  File "/src/ybd/assembly.py", line 130, in assemble
    build(defs, component)
  File "/src/ybd/assembly.py", line 164, in build
    allow_parallel=('build' in build_step))
  File "/src/ybd/sandbox.py", line 196, in run_sandboxed
    env=env, **sandbox_config)
  File "/src/src/sandboxlib/sandboxlib/chroot.py", line 222, in run_sandbox_with_redirection
    exit, out, err = run_sandbox(command, **sandbox_config)
  File "/src/src/sandboxlib/sandboxlib/chroot.py", line 218, in run_sandbox
    raise exception
RuntimeError: Unable to set current working directory: [Errno 2] No such file or directory: 'fhs-dirs.build'
```

Thanks to Paul Martin for reporting in #baserock

**Comment by ssssam** [comment]
This is fixed in the sandboxlib 0.3.1 release.

I have done a full run of `ybd systems/build-system-x86_64.morph` with `SANDBOXLIB_BACKEND=chroot` set in the environment to test that the chroot backend now works.

**Comment by devcurmudgeon** [comment]
> 
> This is fixed in the sandboxlib 0.3.1 release.
> 
> I have done a full run of ybd systems/build-system-x86_64.morph with SANDBOXLIB_BACKEND=chroot set in the environment to test that the chroot backend now works
> 
Excellent! Thanks :)
